src/journal/test/JournalTest.hs

```diff
@@ -41,23 +42,32 @@ prettyFakeJournal :: FakeJournal -> String
 prettyFakeJournal = show . fmap (prettyRunLenEnc . encodeRunLength)
 
 startJournalFake :: FakeJournal
-startJournalFake = FakeJournal Vector.empty 0
+startJournalFake = FakeJournal Vector.empty 0 1
 
 appendBSFake :: ByteString -> FakeJournal -> (FakeJournal, Either AppendError ())
-appendBSFake bs fj@(FakeJournal bss ix)
-  | position < limit =
-    trace (unlines [ "TRACE"
-                   , "ix: " ++ show ix
-                   , "readBytes: " ++ show readBytes
-                   , "unreadBytes: " ++  show unreadBytes
-                   , "readPadBytes: " ++  show readPadBytes
-                   , "unreadPadBytes: " ++ show unreadPadBytes
-                   , "position: " ++ show position
-                   , "limit: " ++ show limit
-                   ])
-    (FakeJournal (Vector.snoc bss bs) ix, Right ())
-  | otherwise           = (fj, Left BackPressure)
+appendBSFake bs fj@(FakeJournal bss ix termCount) =
+  trace (unlines [ "TRACE"
+                 , "ix: " ++ show ix
+                 , "termCount: " ++ show termCount
+                 , "bs: " ++ show (encodeRunLength bs)
+                 , "readBytes: " ++ show readBytes
+                 , "unreadBytes: " ++  show unreadBytes
+                 , "position: " ++ show position
+                 , "limit: " ++ show limit
+                 ]) $
+    if journalLength >= termLen * termCount
+    then (FakeJournal (Vector.snoc bss padd) ix (termCount + 1), Left Rotation)
+    else if position < limit
+    then (FakeJournal (Vector.snoc bss bs) ix termCount, Right ())
+    else (fj, Left BackPressure)
```

Comment:
Shouldn't we check for `BackPressure` first before rotation?

Reply:
You're right, fixed, thanks!